engine/engine_test.go

```diff
@@ -313,7 +314,7 @@ func TestEngine_NotifyPutWithoutListerIsError(t *testing.T) {
 	require.Equal(t, cid.Undef, gotCid)
 }
 
-func TestEngine_NotifyPutThenNotifyRemove(t *testing.T) {
```

Comment:
Is there a test that assets ad chain remains retrievable even if all is removed?

Reply:
added